### DISCUSSION TITLE
What's the best way to build a bot balancing between 429 limits and fast responses?

### DISCUSSION THREAD

**Discussion by FordSmh** [def-studio/telegraph#551] [Ideas]
Single queue is throttling answers to a user, while dealing with other users' messages

**Comment by fabio-ivona** [comment]
Hi, I think we could work on it on one of the next releases

**Reply by FordSmh** [reply]
I was not explicit enough, what I wanted is to get some recommendations and/or Best Practices doc section for building Telegram bots

**Reply by andrey-helldar** [reply]
Just use multiple queues.

My case:

1. Telegram sends a request;
2. The request is written to the database;
3. The record triggers an event that starts Job;
4. Job initializes WebhookHandler;
5. WebhookHandler is executed in multiple threads.

Laravel Horizon + autobalancing

![image](https://github.com/user-attachments/assets/d723f7ac-ed75-454a-9d14-612158a86b16)
